packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -0,0 +1,140 @@
+            'Got %s with POST request to %s with headers %s and json data %s.',
+            r.status_code,
+            url,
+            headers,
+            json
+        )
+        assert r.status_code == 200
+        return r
+
+    def find_container_id(state, app_id):
+        if 'frameworks' in state:
+            for framework in state['frameworks']:
+                if 'tasks' in framework:
+                    for task in framework['tasks']:
+                        if 'id' in task and app_id in task['id']:
+                            if 'statuses' in task and 'container_status' in task['statuses'][0]:
+                                if 'container_id' in task['statuses'][0]['container_status']:
+                                    container_status = task['statuses'][0]['container_status']
+                                    if 'container_id' in container_status:
+                                        if 'value' in container_status['container_id']:
+                                            return container_status['container_id']['value']
+
+    def find_agent_id(state, app_id):
+        if 'frameworks' in state:
+            for framework in state['frameworks']:
+                if 'tasks' in framework:
+                    for task in framework['tasks']:
+                        if 'id' in task and app_id in task['id']:
+                            if 'slave_id' in task:
+                                return task['slave_id']
+
+    def find_agent_hostname(state, agent_id):
+        if 'slaves' in state:
+            for agent in state['slaves']:
+                if 'id' in agent and agent['id'] == agent_id and 'hostname' in agent:
+                    return agent['hostname']
+
+    # Creates and yields the initial ATTACH_CONTAINER_INPUT message, then a data message,
+    # then an empty data chunk to indicate end-of-stream.
+    def _input_streamer(encoder, nested_container_id, input_data):
+        message = {'type': 'ATTACH_CONTAINER_INPUT', 'attach_container_input': {'type': 'CONTAINER_ID'}}
+        message['attach_container_input']['container_id'] = nested_container_id
+        yield encoder.encode(message)
+
+        message = {'type': 'ATTACH_CONTAINER_INPUT', 'attach_container_input': {'type': 'PROCESS_IO'}}
+        message['attach_container_input']['process_io'] = {'type': 'DATA', 'data': {'type': 'STDIN'}}
+        message['attach_container_input']['process_io']['data']['data'] = input_data
+        yield encoder.encode(message)
+
+        # Place an empty string to indicate EOF to the server and push
+        # 'None' to our queue to indicate that we are done processing input.
+        message['attach_container_input']['process_io']['data']['data'] = ''
+        yield encoder.encode(message)
+
+    # Launch a basic marathon app (no image), so we can debug into it!
+    app, test_uuid = get_test_app()
+    test_app_id = 'integration-test-{}'.format(test_uuid)
+    cluster.marathon.deploy_app(app)
```

Comment:
Unfortunately, `deploy_and_cleanup` destroys the app before I can get a hold of the container to debug into it. I need something that only cleans up after I'm done with it, since I need a live running task for the duration of my test. Something like a test Teardown?